File: split/pushnest.pas
procedure pushnest;
begin
  if nestptr > maxneststack then
  begin
    maxneststack := nestptr;
    if nestptr = nestsize then
      overflow(362, nestsize);
  end;
  nest[nestptr] := curlist;
  nestptr := nestptr + 1;
  curlist.headfield := getavail;
  curlist.tailfield := curlist.headfield;
  curlist.pgfield := 0;
  curlist.mlfield := line;
end;

File: cpp/src/pushnest.cpp
#include "pushnest.h"
#include "erreur.h"
#include "noeud.h"

void pushnest(void)
{
	if (nest.size()-1 == nestsize)
		overflow("semantic nest size", nestsize); 
	nest.push_back(curlist);
	tail = head = new LinkedNode;
	prev_graf = 0;
	mode_line = line;
}
